### ISSUE TITLE
Youtube URL not correctly captured

### ISSUE THREAD

**Issue by dinh** [osmoscraft/osmosmemo#4]
To reproduced the bug:
* go to a Youtube URL
* click on another link
* try to bookmark it
* check the captured URL vs the real URL
Tested with FF.

**Comment by chuanqisun** [comment]
I believe YouTube incorrectly put this canonical link on all its videos.
```html
<link rel="canonical" href="https://www.youtube.com/">
```

It should be
```html
<link rel="canonical" href="https://www.youtube.com/watch?v=video_id">
```
I plan to hard code a list of domains that the tool should avoid using the `canonical` tag for determining URL, but that is a sub-optimal workaround. I hope YouTube can fix their bug so we don't have to hack around it. Sadly, I can't find a way to file bugs against YouTube. If anyone knows how to get YouTube's attention on issues like this, please help.

**Comment by chuanqisun** [comment]
Addressed in #7. Released as v2.3.0
The fix is live in Firefox.
Edge and Chrome may take up to a week. 

While fixing this, I noticed when I browse from one video to another, the URL will correctly update but the title won't. This is due to how YouTube failed updating the title in the document as they used JavaScript to handle routing on the client side. There isn't much I can do. Sorry about the inconvenience.

Thank you!
